tests/wasm/runtime/abi.ts

```diff
@@ -312,6 +312,61 @@ export class Abi {
     return ptr
   }
 
+  public writeStoreEntity(value: Store.Entity | null): Pointer {
+    if (value === null) {
+      return 0
+    }
+
+    const entriesPtr = this.writeArray(value.entries, ({ name, value }) => {
+      const ptr = this.allocate(8)
+      this.setWord(ptr, this.writeString(name))
+      this.setWord(ptr + 4, this.writeStoreValue(value))
+      return ptr
+    })
+
+    const ptr = this.allocate(4)
+    this.setWord(ptr, entriesPtr)
+
+    return ptr
+  }
+
+  private writeStoreValue(value: Store.Value): Pointer {
```

Comment:
Maybe a unit test that read and write storage work as expected could be useful by itself (independent of the larger on deposit test).